src/Core/Settings.cpp

```diff
@@ -6430,6 +6430,9 @@ Use local pipeline during distributed INSERT SELECT with parallel replicas
 )", BETA) \
     DECLARE(Milliseconds, parallel_replicas_connect_timeout_ms, 300, R"(
 The timeout in milliseconds for connecting to a remote replica during query execution with parallel replicas. If the timeout is expired, the corresponding replicas is not used for query execution
+)", BETA) \
+    DECLARE(Bool, parallel_replicas_wait_for_unused_replicas, false, R"(
```

Comment:
Do you think that it worth to add a setting just to fix the test? Maybe we can remove this assertion from the test instead?

Reply:
We need to test some functionality, and its execution can't be reach reliably if we cancel `RemoteSource` processors as soon as all read tasks assigned. The setting allows reaching desired code path reliably.

Agree, usage of settings for this purpose can be excessive. We can use fail point mechanism with the same purpose (i.e., as setting for testing) which is not exactly fail point but since it's for testing - should be fine.

Reply:
>We need to test some functionality, and its execution can't be reach reliably if we cancel RemoteSource processors as soon as all read tasks assigned. The setting allows reaching desired code path reliably.

Maybe we can tune parallel replicas that way that it will reach all servers? Either by decreasing `parallel_replicas_mark_segment_size` and some generic sleep, like `sleep_in_send_data_ms`/`sleep_after_receiving_query_ms`, or if this is not enough maybe some generic sleep that will be useful for PR. What do you think?